lib/dialects/abstract/connection-manager.js

```diff
@@ -312,6 +288,36 @@ class ConnectionManager {
     if (!this.dialect.connectionManager.validate) return true;
     return this.dialect.connectionManager.validate(connection);
   }
+
+  _acquire(options) {
+    const connectionPromise = this.pool.acquire(options.priority, options.type, options.useMaster);
+    return this._watchPoolError(connectionPromise).tap(() => { debug('connection acquired'); });
+  }
+
+  _watchPoolError(acquirePromise) {
+    return new Promise((resolve, reject) => {
+      if (acquirePromise.isFulfilled()) {
```

Comment:
For now there isn't, but we are already tied to bluebird and we should use itsl API where required. When we will eventually migrate to `async await` we can change this segment with reset of codebase as well